common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -293,4 +294,45 @@ public static String getJsonPath(String jsonPathWithSource) {
         // Extract the substring from the startIndex to the end of the input string
         return (startIndex != -1) ? jsonPathWithSource.substring(startIndex) : jsonPathWithSource;
     }
+
+    /**
+     * Checks if the given input string matches the dot path format.
+     *
+     * <p>The dot path format is a string that consists of one or more word characters
+     * (letters, digits, or underscores) separated by dots. The string can optionally
+     * start or end with a dot followed by one or more word characters.
+     *
+     * <p>Examples of valid dot path format strings:
+     * <ul>
+     *     <li>"foo"</li>
+     *     <li>"foo.bar"</li>
+     *     <li>"foo.bar.baz"</li>
+     *     <li>"foo.bar.baz.qux"</li>
+     *     <li>".foo"</li>
+     *     <li>"foo."</li>
+     *     <li>".foo.bar"</li>
+     * </ul>
+     *
+     * <p>Examples of invalid dot path format strings:
+     * <ul>
+     *     <li>"foo..bar"</li>
+     *     <li>"."</li>
+     *     <li>".."</li>
+     * </ul>
+     *
+     * @param input the input string to be checked
+     * @return true if the input string matches the dot path format, false otherwise
+     */
+    public static boolean isValidJSONPath(String input) {
```

Comment:
```
Bracket Notation (Useful for property names with special characters):

$.store['book']
$['store']['book'][0]['title']
Array Indexing:

$.store.book[0]
$.store.book[1,2] (Multiple indices)
$.store.book[-1:] (The last element)
$.store.book[0:2] (Slice array from index 0 to 2)
Wildcard (*):

$.store.book[*] (All books)
$.* (All properties at the root level)
Filters (?()):

$.store.book[?(@.price < 10)] (Books cheaper than 10)
$.store.book[?(@.author == 'J.K. Rowling')] (Books authored by J.K. Rowling)
Recursive Descent (..):

$..author (Find all authors, no matter where they are in the document)
$..book[?(@.price > 15)] (Find books with a price greater than 15 anywhere in the document)
Union ([,]):

$.store.book[0,1] (First and second books)
$['store','warehouse'] (Select multiple properties at the same level)
Combination of Wildcard and Filters:

$..book[?(@.price > 20)].title (Get the titles of books costing more than 20)
```


Are we going to validate these kinds of json path too?

OR is it just a simple dot path format? Either reflect documentation or the method name accordingly.

Reply:
we need to support all kinds of json, that's why is called `isValidJsonPath`